py/torch_tensorrt/dynamo/_refit.py

```diff
@@ -0,0 +1,364 @@
+            if not isinstance(
+                submodule, (PythonTorchTensorRTModule, TorchTensorRTModule)
+            ):
+                continue
+            settings = submodule.settings
+
+    if settings.debug:
+        set_log_level(logger.parent, logging.DEBUG)
+
+    if not isinstance(inputs, collections.abc.Sequence):
+        inputs = [inputs]
+
+    # Prepare torch_trt inputs
+    inputs = prepare_inputs(inputs)
+    device = to_torch_tensorrt_device(settings.device)
+    runtime = trt.Runtime(TRT_LOGGER)
+    if not isinstance(new_weight_module, ExportedProgram):
+        raise AssertionError(
+            f"Input graph should be an ExportedProgram but got type {type(new_weight_module)}"
+        )
+    new_weight_module = new_weight_module.run_decompositions(
+        get_decompositions(settings.enable_experimental_decompositions)
+    )
+    new_gm = new_weight_module.module()
+    logger.debug("Input graph: " + str(new_gm.graph))
+    # Apply lowering on the graph module
+    torch_inputs = get_torch_inputs(inputs, device)
+    new_gm = apply_lowering_passes(new_gm, torch_inputs)
+
+    logger.info("Compilation Settings: %s\n", settings)
+
+    # Set torch-executed ops
+    CONVERTERS.set_disallowed_targets(settings.torch_executed_ops)
+
+    # If specified, try using the fast partitioner and fall back to the global one on failure
+    if settings.use_fast_partitioner:
+        try:
+            new_partitioned_module, supported_ops = partitioning.fast_partition(
+                new_gm,
+                verbose=settings.debug,
+                min_block_size=settings.min_block_size,
+                torch_executed_ops=settings.torch_executed_ops,
+            )
+        except torch.fx.passes.splitter_base.FxNetSplitterInternalError:
+            logger.error(
+                "Partitioning failed on the subgraph with fast partition. See trace above. "
+                + "Retrying with global partition.",
+                exc_info=True,
+            )
+
+            fast_partitioner_failed = True
+            settings.use_fast_partitioner = False
+
+    if not settings.use_fast_partitioner:
+        new_partitioned_module, supported_ops = partitioning.global_partition(
+            new_gm,
+            verbose=settings.debug,
+            min_block_size=settings.min_block_size,
+            torch_executed_ops=settings.torch_executed_ops,
+        )
+
+    if inline_module:
+        # Preprocess the partitioned module to be in the same format as the inline module
+        inline_torch_modules(new_partitioned_module)
+        new_partitioned_module.delete_all_unused_submodules()
+
+    # Check whether two modules have the same subcomponents
+    # 1. Check the number of partitions and name
+    if inline_module:
+        assert {sm[0] for sm in new_partitioned_module.named_children()} == set(
+            compiled_submodules_map.keys()
+        ), "The compiled module is incompatible with the new module!"
```

Comment:
Can we flip the wording here to be more clear? 
Ex. `New weights module is not compatible with previously compiled Torch-TensorRT module`